.github/workflows/release.yml

```diff
@@ -153,3 +153,35 @@ jobs:
                 version: "${{ env.VERSION }}",
               }
             });
+
+  pypi-publish:
+    runs-on:
+      group: databricks-deco-testing-runner-group
+      labels: ubuntu-latest-deco
+
+    needs: goreleaser
+
+    environment: release
```

Comment:
Why does this need access to the release environment token?

Reply:
It's specified in trusted publishing settings in PyPi package deployment settings to avoid cases when package gets accidentially published

Reply:
Please add a comment to document this assertion.

Since the workflow doesn't use secrets, someone may remove this if they don't know about the assertion made by PyPI for OIDC purposes.

Reply:
Clarified in comments

Reply:
Can you include that this is part of the OIDC assertion in PyPI?

Docs: https://docs.pypi.org/trusted-publishers/creating-a-project-through-oidc/

Reply:
@pietern fixed, please take a look